### DISCUSSION TITLE
evalsha: how to calculate sh1 digest without going to redis server?

### DISCUSSION THREAD

**Discussion by y2241** [redis/lettuce#2293] [General]
If I have a lua script and I want to know its sha digest value without using scriptLoad(), how can I do that?
I tried to use Apache DigestUtils but its digest hash value is different from what I got by calling  scriptLoad(). The values are the same on my local, but when running this in the pipeline in the cloud, their values are different.
If I could get the SHA1 string of the lua script, I can then directly call evalsha(digest), so that I don't have to call scriptLoad() for each jvm, as I have many jvms using the same redis instance.
Currently each application instance (when it is started), trying to put the lua script into redis cache by calling scriptLoad(), which is unnecessary. But the application does not know if the script is already on redis cache because it does not know the SHA1.

Any suggestion is highly appreciated.

**Comment by mp911de** [comment]
You can use `Base16.digest(…)` or look up any other Java-based SHA1 computation tutorial.
